server/deskroom/knowledge_base/endpoints.py

```diff
@@ -19,8 +29,59 @@ async def get_knowledge_base(
 ) -> list[KnowledgeBase]:
     response = (
         await supabase.table("knowledge_base")
-        .select("*")
+        .select("*, organizations(company_info_policy)")
         .eq("org_key", item.organization_key)
         .execute()
     )
+
     return response.data
+
+
+@router.post("/upload/{org_key}")
+async def make_knowledge_base(
+    org_key: str,
+    file: UploadFile = File(...),
+    supabase: AsyncClient = Depends(create_supabase_async_client),
+) -> list[KnowledgeBase]:
+    questions = []
+    answers = []
+
+    raw_df = pd.read_excel(
+        file.file.read(), engine="openpyxl", sheet_name="Message data"
+    )
+
+    supabase_response = await (
+        supabase.table("organizations").select("*").eq("key", org_key).execute()
+    )
+    org_info = supabase_response.data[0]
+
+    raw_df = raw_df.dropna()
+    processed_df = await process_raw_file(raw_df)
+
+    discovery_str = await generate_discovery_string(processed_df)
+    company_policy = await create_policy(discovery_str)
+    chat_ids = list(processed_df["chatId"].unique())
+    for chat_id in chat_ids:
+        try:
+            qa_string = await generate_qa_string(processed_df, chat_id)
+            discovered = await create_qa(company_policy, qa_string)
+            discovered_ = literal_eval(discovered)
+            for qa in list(discovered_.values()):
+                questions.append(qa["Question"])
+                answers.append(qa["Answer"])
+        except BaseException:
```

Comment:
```suggestion
        except (ValueError, KeyError):
```
BaseException은 모든 python 에러들의 상위 클래스입니다. 이렇게 되면 이 라인에서 value가 잘못 된건지, 단순한 runtime error인지 크리티컬한 에러인지 모르게 될 수 있습니다. 

Golden rule은
1. try-except를 최대한 적은 라인으로 유지하세요.
2. 에러 핸들링하는 케이스는 최대한 좁게 유지하세요. 
이 케이스에는 그냥 디버깅하기 편하라고 해둔거 같아서 에러 스코프를 최대한 좁혀주세요.